From 1e3bb1410bee9fc20d441380ee8b0fc8ddf390a3 Mon Sep 17 00:00:00 2001
From: Marsell Kukuljevic <marsell@joyent.com>
Date: Wed, 22 Nov 2017 00:25:27 +1300
Subject: [PATCH] PUBAPI-1462: Tests not terminating when run without
 SDC_SETUP_TESTS Reviewed by: Pedro P. Candel <pedro@joyent.com>

---
 .gitignore |  1 +
 lib/app.js | 12 ++++++++++++
 2 files changed, 13 insertions(+)

diff --git a/.gitignore b/.gitignore
index 3d47f02..8cd521a 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
+*.swp
 build
 *.node
 *.pdf
diff --git a/lib/app.js b/lib/app.js
index 2d150fd..11802d1 100644
--- a/lib/app.js
+++ b/lib/app.js
@@ -732,6 +732,18 @@ module.exports = {
                         clients.mahi.close();
                     }
 
+                    if (clients.cns) {
+                        clients.cns.close();
+                    }
+
+                    if (clients.volapi) {
+                        clients.volapi.close();
+                    }
+
+                    process._getActiveHandles().forEach(function (h) {
+                        return h.unref && h.unref();
+                    });
+
                     clients.ufds.close(function (err) {
                         if (err) {
                             throw err;
-- 
2.21.0

